tools/deny.toml

```diff
@@ -0,0 +1,207 @@
+# This template contains all of the possible sections and their default values
+
+# Note that all fields that take a lint level have these possible values:
+# * deny - An error will be produced and the check will fail
+# * warn - A warning will be produced, but the check will not fail
+# * allow - No warning or error will be produced, though in some cases a note
+# will be
+
+# The values provided in this template are the default values that will be used
+# when any section or field is not specified in your own configuration
+
+# If 1 or more target triples (and optionally, target_features) are specified,
+# only the specified targets will be checked when running `cargo deny check`.
+# This means, if a particular package is only ever used as a target specific
+# dependency, such as, for example, the `nix` crate only being used via the
+# `target_family = "unix"` configuration, that only having windows targets in
+# this list would mean the nix crate, as well as any of its exclusive
+# dependencies not shared by any other crates, would be ignored, as the target
+# list here is effectively saying which targets you are building for.
+targets = [
+    # The triple can be any string, but only the target triples built in to
+    # rustc (as of 1.40) can be checked against actual config expressions
+    #{ triple = "x86_64-unknown-linux-musl" },
+    # You can also specify which target_features you promise are enabled for a
+    # particular target. target_features are currently not validated against
+    # the actual valid features supported by the target architecture.
+    #{ triple = "wasm32-unknown-unknown", features = ["atomics"] },
+]
+
+# This section is considered when running `cargo deny check advisories`
+# More documentation for the advisories section can be found here:
+# https://embarkstudios.github.io/cargo-deny/checks/advisories/cfg.html
+[advisories]
+# The path where the advisory database is cloned/fetched into
+db-path = "~/.cargo/advisory-db"
+# The url(s) of the advisory databases to use
+db-urls = ["https://github.com/rustsec/advisory-db"]
+# The lint level for security vulnerabilities
+vulnerability = "deny"
+# The lint level for unmaintained crates
+unmaintained = "deny"
+# The lint level for crates that have been yanked from their source registry
+yanked = "deny"
+# The lint level for crates with security notices. Note that as of
+# 2019-12-17 there are no security notice advisories in
+# https://github.com/rustsec/advisory-db
+notice = "warn"
+# A list of advisory IDs to ignore. Note that ignored advisories will still
+# output a note when they are encountered.
+ignore = [
+    "RUSTSEC-2021-0127", # serde_cbor unmaintained; only used in tests
+]
```

Comment:
Should we still keep CBOR in the tests, then? It does not seem to be particularly popular vs the similar MessagePack, despite being "standard".

The serde documentation currently links to [`cborium`](https://github.com/enarx/ciborium), which *is* maintained, if very starved of attention, as the CBOR implementation instead of `serde_cbor`.

Reply:
I asked myself the same question while looking at this. I also had a quick look at both of the suggested replacements [`minicbor`](https://docs.rs/minicbor) and [`cborium`](https://docs.rs/ciborium/) (docs.rs links). The former seemed to have a focus on embedded systems and needed separate traits to be implemented, and for the latter I didn't find a quick equivalent for `to_vec()` in the tests, and didn't want to spend too much time on it.

I'm personally totally OK with dropping the CBOR tests. If we find another popular format in the future, we could always add that, but I think test coverage is already pretty good anyway. Maybe @Waridley can say something about this?